src/common/components/ui/formik/validationSchemas.ts

```diff
@@ -26,9 +26,8 @@ const NAME_START_END_REGEX = /^[a-z0-9](.*[a-z0-9])?$/;
 const NAME_CHARS_REGEX = /^[a-z0-9-.]*$/;
 const CLUSTER_NAME_START_END_REGEX = /^[a-z0-9](.*[a-z0-9])?$/;
 // NOTE: based on https://github.com/openshift/assisted-service/blob/master/internal/cluster/validations/validations.go#L32
-const CLUSTER_NAME_REGEX =
-  /^[.-a-z0-9]?(([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)*)[.-a-z0-9]?$/;
-const CLUSTER_NAME_NO_DOT_REGEX = /^[a-z0-9-]*$/;
+const CLUSTER_NAME_NO_DOT_REGEX =
+  /^[a-z0-9]?(([a-z0-9]([-a-z0-9]*[a-z0-9])?([a-z0-9]([-a-z0-9]*[a-z0-9])?)*)*)[a-z0-9]?$/;
```

Comment:
When the dots are removed, this expression is way too complicated.

Without dots, we can check for kind of following one (not tested!!):
`/^[a-z0-9]?[a-z0-9-]*$/`

(Must start at an alphanumeric, followed by any count of alphanumerics or dashes. Can end at dash.)

Reply:
I agree it looks too complicated, let me see if I can refine it.
It should not be allowed to end in a dash, as there is a separate check for it (via `CLUSTER_NAME_START_END_REGEX`).
![name](https://user-images.githubusercontent.com/829045/208849580-fea88838-b834-4dc0-a0df-ac1e12de1885.png)

Reply:
ok, then. I wasn't sure. Should I play with putting the expression together?

Reply:
@mareklibra This last comment made me think, we don't need the main RegExp to cover all cases, given that we're using `CLUSTER_NAME_START_END_REGEX` to identify the start and end of the name.

Therefore the "main" RegExp only cares about defining the allowed characters. I put back the simple RegExp expression that @jtomasek had inially written, only I renamed it for clarity.

See how it works now:
https://user-images.githubusercontent.com/829045/208850959-9dcd59f7-723b-4547-b122-bd1096fcb1f6.mp4

Reply:
You are right, this is better when we already have it split.